src/multi_select.coffee

```diff
@@ -101,12 +114,33 @@ Ember.Widgets.MultiSelectComponent = Ember.Widgets.SelectComponent.extend
     @set 'values', Ember.A [] unless @get('values')
 
   deletePressed: (event) ->
-    if event.target.selectionStart == 0
+    if event.target.selectionStart is 0 and event.target.selectionEnd is 0
       @removeSelectItem(@get('selections.lastObject'))
+      event.preventDefault()
 
   removeSelectItem: (item) ->
+    # set the focus back to the searchView because this item will be removed
+    currentDropDownState = @get('showDropdown')
+    @focusTextField()
+    @send 'hideDropdown' unless currentDropDownState
     @get('selections').removeObject item
 
+  escapePressed: (event) ->
+    if @get('showDropdown')
+      @focusTextField()
+      @send 'hideDropdown'
+      event.preventDefault()
+
+  enterPressed: (event) ->
+    @_super(event)
+    @focusTextField()
+    @set('showDropdown', yes) unless @get('showDropdown')
```

Comment:
if it's already `yes`, there should be no harm setting it to `yes`, right? Can we remove the `unless`?